Что выведет программа? Объяснить вывод программы.

```
package main
 
type customError struct {
     msg string
}
 
func (e *customError) Error() string {
    return e.msg
}
 
func test() *customError {
     {
         // do something
     }
     return nil
}
 
func main() {
    var err error
    err = test()
    if err != nil {
        println("error")
        return
    }
    println("ok")
}
```

Получим ту же самую проблему, что и пару примеров назад, в интерфейс ошибки записываем *customError, поэтому itab не будет nil, а значит и err не будет nil 